### ISSUE TITLE
Colima unable to start - stuck on sshfs binary install

### ISSUE THREAD

**Issue by Oakwhisper** [abiosoft/colima#871]
### Description

After updating to 0.6.2, I am unable to start Colima. It hangs on this step: 
```[hostagent] Waiting for the essential requirement 3 of 5: "sshfs binary to be installed"```
and eventually fails like this:
```
> did not receive an event with the "running" status
FATA[0600] error starting vm: error at 'starting': exit status 1
```

This happens after a reinstall of Colima and Lima. I have cleared the Colima cache as well as restarted my computer. 

### Version

colima version 0.6.2
git commit: 22d7e5fbc86d5b8e3b27065a762800bc7960a0ff
limactl version 0.18.0
qemu-img version 8.1.2
Copyright (c) 2003-2023 Fabrice Bellard and the QEMU Project developers

### Operating System

- [ ] macOS Intel <= 13 (Ventura)
- [ ] macOS Intel >= 14 (Sonoma)
- [ ] Apple Silicon <= 13 (Ventura)
- [X] Apple Silicon >= 14 (Sonoma)
- [ ] Linux

### Output of `colima status`

FATA[0000] error retrieving current runtime: empty value

### Reproduction Steps

1. brew uninstall colima
2. brew uninstall lima
3. brew install colima
4. colima start

### Expected behaviour

Colima to start. It has previously on this device.

### Additional context

My company uses proxies, but I have tried to disable them and use a non-work network to perform the install and startup. Previously (version 0.5.2) I couldn't start Colima unless I was off vpn and had set the proxies accordingly. Now even that won't work.

**Comment by tiilikainen** [comment]
Below is information and a workaround I got from my colleague. I take no credit.

Colima switched from Alpine to Ubuntu as its underlying host VM OS starting with v6.0. If you tail the logs during startup (`~/.colima/_lima/colima/serialv.log`), you will notice that `apt` tries to run and fails due to a network connectivity issue (unknown what is causing it at this point).

Here is a temporary workaround. There's no need to uninstall `colima` a second time. Start (or restart) Colima, and wait until you see the `sshfs binary to be installed` message again. At that point, open up a new terminal window and issue the following command:

```bash
colima ssh -- sh -c "sudo apt-get update && sudo apt-get install -y -o Dpkg::Options::=--force-confdef -o Dpkg::Options::=--force-confnew sshfs"
```

This forces a package update and the installation of `sshfs`, which will allow the boot process to continue. You need only do this once per install of Colima.

**Comment by Oakwhisper** [comment]
Unfortunately it seems that probably isn't my issue. My serialv.log is empty and when I tried to run the command you listed, I get the following:
```
Hit:1 http://ports.ubuntu.com/ubuntu-ports mantic InRelease
Hit:2 http://ports.ubuntu.com/ubuntu-ports mantic-updates InRelease
Hit:3 http://ports.ubuntu.com/ubuntu-ports mantic-backports InRelease
Hit:4 http://ports.ubuntu.com/ubuntu-ports mantic-security InRelease
Reading package lists... Done
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package sshfs
FATA[0000] exit status 100
```

**Comment by tiilikainen** [comment]
Hmm. Can you paste your `~/.docker/config.json`? Do you have proxies defined there?

**Comment by Oakwhisper** [comment]
No proxies defined there, just auth keys
```
{
	"auths": {
		"[COMPANY REGISTRY]": {
			"auth": [REDACTED]
		},
		"us-central1-docker.pkg.dev": {
			"auth": [REDACTED]
		}
	}
}
```

**Comment by tiilikainen** [comment]
I use `cntlm` as an authentication proxy. Here is my `docker.json`:

```json
{
        "proxies": {
                "default": {
                        "httpProxy": "http://host.lima.internal:3128",
                        "httpsProxy": "http://host.lima.internal:3128",
                        "noProxy": "127.0.0.1,localhost,host.lima.internal,.local,.internal,169.254.169.254,*.mycompany.com"
                }
        },
        "currentContext": "colima"
}
```

**Comment by tiilikainen** [comment]
I can confirm that 0.6.5 does _not_ fix the `sshfs` issue:

```
INFO[0000] editing in vim from $EDITOR environment variable
INFO[0013] starting colima
INFO[0013] runtime: docker
INFO[0013] creating and starting ...                     context=vm
> [hostagent] Waiting for the essential requirement 3 of 5: "sshfs binary to be installed"
> [hostagent] Waiting for the essential requirement 3 of 5: "sshfs binary to be installed"
> [hostagent] Waiting for the essential requirement 3 of 5: "sshfs binary to be installed"
> [hostagent] Waiting for the essential requirement 3 of 5: "sshfs binary to be installed"
> [hostagent] Waiting for the essential requirement 3 of 5: "sshfs binary to be installed"
> [hostagent] Waiting for the essential requirement 3 of 5: "sshfs binary to be installed"
```

**Comment by ivanint** [comment]
sshfs doesn't install, docker pull doesn't work if using proxy even if env variables are defined and seen in colima ssh

**Comment by abiosoft** [comment]
> sshfs doesn't install, docker pull doesn't work if using proxy even if env variables are defined and seen in colima ssh. Had to downgrade to v0.5.5

This is currently being worked on to ensure that no package downloads are required during initial startup, similar to v0.5.x.

**Comment by abiosoft** [comment]
This should be fixed in v0.6.7.
